contracts/core/interfaces/IBaseStrategy.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.19;
+
+import {IAllo} from "./IAllo.sol";
+
+interface IBaseStrategy {
+    /// ======================
+    /// ======= Errors =======
+    /// ======================
+
+    /// @notice Error when the sender is not authorized
+    error BaseStrategy_UNAUTHORIZED();
+
+    /// @notice Error when the strategy is not initialized
+    error BaseStrategy_NOT_INITIALIZED();
+
+    /// @notice Error when the strategy is already initialized
+    error BaseStrategy_ALREADY_INITIALIZED();
+
+    /// @notice Error when the pool ID is invalid
+    error BaseStrategy_INVALID_POOL_ID();
+
+    /// ======================
+    /// ======= Events =======
+    /// ======================
+
+    /// @notice Emit after recipients get registered
+    /// @param _recipient The address of the recipient
+    /// @param _data The data to use to register the recipient
+    event Registered(address indexed _recipient, bytes _data);
+
+    /// @notice Emits after recipients got allocated amount
+    /// @param _recipient The address of the recipient
+    /// @param _sender The address of the sender
+    /// @param _amount The amounts to allocate
+    /// @param _data The data to use to allocate to the recipient
+    event Allocated(address indexed _recipient, address indexed _sender, uint256 _amount, bytes _data);
+
+    /// @notice Emit after recipients got distributed
+    /// @param _recipient The ID of the recipient
+    /// @param _data The data to use to distribute to the recipients
+    event Distributed(address indexed _recipient, bytes _data);
+
+    /// @notice Emits after a withdrawal is made
+    /// @param _token The address of the token
+    /// @param _amount The amount to withdraw
+    /// @param _recipient The address to withdraw to
+    event Withdrew(address indexed _token, uint256 _amount, address _recipient);
+
+    /// @notice Emits when the pool is initialized
+    /// @param poolId The ID of the pool
+    /// @param data The data passed to the 'initialize' function
+    event Initialized(uint256 poolId, bytes data);
+
+    /// ======================
+    /// ======= Views ========
+    /// ======================
+    /// @notice Getter for the address of the Allo contract.
+    /// @return _allo The 'Allo' contract
+    function getAllo() external view returns (IAllo _allo);
+
+    /// @notice Getter for the 'poolId' for this strategy.
+    /// @return _poolId The ID of the pool
+    function getPoolId() external view returns (uint256 _poolId);
+
+    /// @notice Checks the amount of tokens in the pool.
+    /// @return _poolAmount The balance of the pool
+    function getPoolAmount() external view returns (uint256 _poolAmount);
+
+    /// @notice Gets the timestamp of the phase at the given index.
+    /// @param _index The index of the phase
+    function getTimestampPhase(uint256 _index) external view returns (uint256 _timestamp);
```

Comment:
What's this about ?

Reply:
yea not really needed, now that i think about it.